core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -99,6 +104,56 @@ public S3RestServiceHandler(@Context ServletContext context) {
         (FileSystem) context.getAttribute(ProxyWebServer.FILE_SYSTEM_SERVLET_RESOURCE_KEY);
   }
 
+  private static String getUserFromAuthorization(String authorization) {
+    if (authorization == null) {
+      return "";
+    }
+
+    // The valid pattern for Authorization is "AWS <AWSAccessKeyId>:<Singature>"
+    int spaceIndex = authorization.indexOf(' ');
+    if (spaceIndex == -1) {
+      return "";
+    }
+    String stripped = authorization.substring(spaceIndex + 1);
+
+    int colonIndex = stripped.indexOf(':');
+    if (colonIndex == -1) {
+      return "";
+    }
+
+    return stripped.substring(0, colonIndex);
+  }
+
+  /**
+   * @summary lists all buckets owned by you
+   * @param authorization header parameter authorization
+   * @return the response object
+   */
+  @GET
+  @Path("/")
+  public Response listAllMyBuckets(@HeaderParam("Authorization") String authorization) {
+
+    return S3RestUtils.call("", new S3RestUtils.RestCallable<ListAllMyBucketsResult>() {
+      @Override
+      public ListAllMyBucketsResult call() throws S3Exception {
+        String user = getUserFromAuthorization(authorization);
+
+        List<URIStatus> objects;
+        try {
+          objects = mFileSystem.listStatus(new AlluxioURI("/"));
```

Comment:
What exactly is the concept of a bucket in the Alluxio proxy?

Reply:
Bucket in the concept of alluxio is the most outer folder. This means that if there are files in the outer folder. There will be no way to view it using the S3 API.

Reply:
We may need to update the behavior in the future.

Another possibility is that we always have a "root" bucket which just holds the files under root, and the directories under root are just separate buckets.

In any case, I don't think there is a clear spec for this currently, so we may have to be flexible in the future.